Sources/NIOPosix/SocketChannel.swift

```diff
@@ -263,6 +263,7 @@ final class ServerSocketChannel: BaseSocketChannel<ServerSocket> {
             // It's important to call the methods before we actually notify the original promise for ordering reasons.
             self.becomeActive0(promise: promise)
         }.whenFailure{ error in
+            self.close0(error: error, mode: .all, promise: nil)
```

Comment:
Upon bind failing, the `lifecycleManager` seems to remain in the `preRegistered` state. thus we are transiting to the `closed` state here.